src/server/pkg/sync/sync.go

```diff
@@ -0,0 +1,127 @@
+// Package sync provides utility functions similar to `git pull/push` for PFS
+package sync
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	pachclient "github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+
+	protostream "go.pedge.io/proto/stream"
+	"golang.org/x/sync/errgroup"
+)
+
+// Pull clones an entire repo at a certain commit
+//
+// root is the local path you want to clone to
+// commit is the commit you want to clone
+// shard and diffMethod get passed to ListFile and GetFile. See documentations
+// for those functions for details on these arguments.
+func Pull(ctx context.Context, client pfs.APIClient, root string, commit *pfs.Commit, diffMethod *pfs.DiffMethod, shard *pfs.Shard) error {
```

Comment:
I like the name `Pull` better because it's the opposite of `Push`.  But yeah I think exposing it in `pachctl` makes sense.  I'd call the command `pachctl clone` since this is most similar to `git clone` but `pachctl checkout` would be fine too.